mantidproject/mantid
# Fix crash importing mslice cli in workbench

**Description of work.**

Forces update to algorithm selector onto GUI thread and more methods of matplotlib figure manager onto GUI thread.

See final commit message for further details.

**To test:**

In workbench running

```python
import mslice.cli
```

no longer crashes.

There are also updates to the figuremanager to ensure that all future methods are run on the QApplication thread. Try any mantid plotting examples from https://docs.mantidproject.org/nightly/plotting/index.html#plotting in the script editor and they should not crash. This is best done on a Windows debug build so it loudly complains if anything is wrong.

Refs #29061

Note this does not yet fix the crash as mslice also needs updates using the new methods provided here.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
